### DISCUSSION TITLE
undefined get_int64_in_string() at ondemand::document?

### DISCUSSION THREAD

**Discussion by ghost** [simdjson/simdjson#1933] [Q&A] [answered]
I try this code:

    simdjson::ondemand::parser parser;
    auto json = simdjson::padded_string(param.input);
    auto dataDoc = parser.iterate(json);

    int64_t result;
    dataDoc.get_int64_in_string().get(result);

and got this error message:

    ValueStringToInt64.cpp:36:13: error: no member named 'get_int64_in_string' in 'simdjson::simdjson_result<fallback::ondemand::document>'
        dataDoc.get_int64_in_string().get(result);
        ~~~~~~~ ^

What's went wrong? According to API documentation `get_int64_in_string()` should be a member of `ondemand::document`.

BTW: i use this compiler:

    Debian clang version 11.0.1-2
    Target: x86_64-pc-linux-gnu
    Thread model: posix
    InstalledDir: /usr/bin

**Comment by lemire** [comment, marked as answer]
The error comes in because dataDoc is not a document but simdjson_result<document>. I will release a patch to make this more convenient. Meanwhile, you can just do...

```
simdjson::ondemand::parser parser;
auto json = simdjson::padded_string(param.input);
simdjson::ondemand::document dataDoc = parser.iterate(json);

int64_t result;
dataDoc.get_int64_in_string().get(result);
```
and it should work.

**Reply by ghost** [reply]
Oh, i see. `dataDoc.value().get_int64_in_string()` works as well. So i should always use `value()` with `simdjson_result<>` data types if i want to be an the safe side.

Thanks a lot for the quick help.

**Reply by lemire** [reply]
Well, you did find a flaw. What you did should work. I will patch simdjson very soon so that your code (as it was) will work. You did nothing wrong, the error is on our side. But yes, you have workarounds.
